daemon/export_test.go

```diff
@@ -214,10 +215,12 @@ func MockSnapstateInstallPathMany(f func(context.Context, *state.State, []*snap.
 	}
 }
 
-func MockSnapstateEnforceSnaps(f func(ctx context.Context, st *state.State, validationSets []string, validErr *snapasserts.ValidationSetsValidationError, userID int) ([]*state.TaskSet, []string, error)) func() {
-	r := testutil.Backup(&assertstateTryEnforceValidationSets)
-	snapstateEnforceSnaps = f
-	return r
+func MockSnapstateResolveValSetEnforcementError(f func(context.Context, *state.State, *snapasserts.ValidationSetsValidationError, map[string]*asserts.ValidationSet, int) ([]*state.TaskSet, []string, error)) func() {
+	old := snapstateResolveValSetEnforcementError
+	snapstateResolveValSetEnforcementError = f
```

Comment:
you can use testutil.Backup()

Reply:
I thought about it but just returning the function always seems more transparent to me =/